### PR TITLE
Acquire IdealStateUpdaterLock for PinotLLCRealtimeSegmentManager IdealState update

### PR DESCRIPTION
**Pull request by xiangfu0**
Acquire IdealStateUpdaterLock for PinotLLCRealtimeSegmentManager IdealState update.

Why the PR:
We observed a lot of race condition and timeout for a large realtime table with SegmentRefreshTask running.
The issue here is both realtime segment commit and SegmentRefreshTask will do large segments update in the idealState.

This PR tries to let them acquire the same lock to avoid race conditions between.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -1979,12 +1979,14 @@ public void setExistingTableConfig(TableConfig tableConfig, int expectedVersion)
     // Update IdealState replication
     IdealState idealState = _helixAdmin.getResourceIdealState(_helixClusterName, tableNameWithType);
     String replicationConfigured = Integer.toString(tableConfig.getReplication());
-    if (!idealState.getReplicas().equals(replicationConfigured)) {
-      HelixHelper.updateIdealState(_helixZkManager, tableNameWithType, is -> {
-        assert is != null;
-        is.setReplicas(replicationConfigured);
-        return is;
-      }, RetryPolicies.exponentialBackoffRetryPolicy(5, 1000L, 1.2f));
+    synchronized (getIdealStateUpdaterLock(tableNameWithType)) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java:1982]: Move this inside the if check

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java [hunk 2]
```diff
@@ -1756,16 +1760,18 @@ private IdealState updatePauseStateInIdealState(String tableNameWithType, boolea
...
+      IdealState updatedIdealState = HelixHelper.updateIdealState(_helixManager, tableNameWithType, idealState -> {
+        ZNRecord znRecord = idealState.getRecord();
+        znRecord.setSimpleField(PAUSE_STATE, pauseState.toJsonString());
+        // maintain for backward compatibility
+        znRecord.setSimpleField(IS_TABLE_PAUSED, Boolean.valueOf(pause).toString());
+        return new IdealState(znRecord);
+      }, RetryPolicies.noDelayRetryPolicy(3));
+      LOGGER.info("Set 'pauseState' to {} in the Ideal State for table {}. "
+          + "Also set 'isTablePaused' to {} for backward compatibility.", pauseState, tableNameWithType, pause);
+      return updatedIdealState;
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java:1773]: (nit) Move them out of the synchronized block

**Review by Jackie-Jiang — APPROVED** [review]

**Comment by tibrewalpratik17** [comment]
hey @xiangfu0 can you share some details in the description on the issue(s) which led to doing this? We can refer it later if we face similar scenario and cherry-pick this PR.

**Comment by xiangfu0** [comment]
> hey @xiangfu0 can you share some details in the description on the issue(s) which led to doing this? We can refer it later if we face similar scenario and cherry-pick this PR.

added PR description

**Pull request merged**